packages/venia-ui/lib/components/CheckoutPage/GuestSignIn/guestSignIn.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { bool, func, shape, string } from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { useGuestSignIn } from '@magento/peregrine/lib/talons/CheckoutPage/GuestSignIn/useGuestSignIn';
+
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+import CreateAccount from '@magento/venia-ui/lib/components/CreateAccount';
+import ForgotPassword from '@magento/venia-ui/lib/components/ForgotPassword';
+import LinkButton from '@magento/venia-ui/lib/components/LinkButton';
+import SignIn from '@magento/venia-ui/lib/components/SignIn';
+import defaultClasses from './guestSignIn.css';
+
+const GuestSignIn = props => {
+    const { isActive, toggleActiveContent } = props;
+
+    const talonProps = useGuestSignIn({ toggleActiveContent });
+    const {
+        handleBackToCheckout,
+        toggleCreateAccountView,
+        toggleForgotPasswordView,
+        view
+    } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const rootClass = isActive ? classes.root : classes.root_hidden;
+
+    let content;
+    if (view === 'SIGNIN') {
```

Comment:
Minor but I'd love if these "view modes" constants were also exported from the talon so we could re-use them here in the component.

Reply:
I pulled most of this logic from `accountMenu`, which I think was you. Can you clarify what you mean by this? I feel like we discussed this in team time awhile ago, and something similar to this idea was pulled out; I think because there wasn't much different between a string literal and named constant.

Reply:
Now that you mention it yeah I think we did talk about it. No worries, it's definitely a personal preference.